src/app/frontend/deploy/deploy.state.js

```diff
@@ -26,6 +26,22 @@ export default function stateConfig($stateProvider) {
     controller: DeployController,
     controllerAs: 'ctrl',
     url: '/deploy',
+    resolve: {
+      namespaces: resolveNamespaces,
+    },
     templateUrl: 'deploy/deploy.html',
   });
+
+  /**
+   * Resolves namespaces for the deploy view.
+   *
+   * @param {!angular.$resource} $resource
+   * @ngInject
+   */
+  function resolveNamespaces($resource) {
+    /** @type {!angular.Resource<!Array<!string>>} */
+    let resource = $resource('/api/namespace');
```

Comment:
Style nit: primitives in JSDoc are always non-null, thus instead of `!string` write `string`.